tst/EnergyPlus/unit/ZoneEquipmentManager.unit.cc

```diff
@@ -4406,3 +4406,135 @@ TEST_F(EnergyPlusFixture, ZoneAirMassFlowBalance_ZoneMixingInfiltrationFlowsFlag
+    state->dataSize->ZoneSizThermSetPtHi(1) = 24;
+    state->dataSize->ZoneSizThermSetPtLo(1) = 22;
+    state->dataGlobal->HourOfDay = 1;
+    state->dataGlobal->NumOfTimeStepInHour = 1;
+    state->dataGlobal->TimeStep = 1;
+    state->dataSize->NumTimeStepsInAvg = 1;
+    state->dataHVACGlobal->FracTimeStepZone = 1;
+    state->dataEnvrn->TotDesDays = 1;
+    state->dataZoneEquipmentManager->NumOfTimeStepInDay = 1;
+    state->dataSize->DesDayWeath.allocate(1);
+    state->dataSize->DesDayWeath(1).Temp.allocate(1);
+    state->dataSize->DesDayWeath(1).Temp(1) = 28;
+    state->dataZoneEquipmentManager->AvgData.allocate(1);
+
+    state->dataSize->ZoneSizing(1, 1).allocateMemberArrays(1);
+    state->dataSize->CalcZoneSizing(1, 1).allocateMemberArrays(1);
+    state->dataSize->CalcFinalZoneSizing(1).allocateMemberArrays(1);
+    state->dataSize->FinalZoneSizing(1).allocateMemberArrays(1);
+
+    UpdateZoneSizing(*state, DataGlobalConstants::CallIndicator::DuringDay);
+    UpdateZoneSizing(*state, DataGlobalConstants::CallIndicator::EndDay);
+    state->dataGlobal->isPulseZoneSizing = true;
+    UpdateZoneSizing(*state, DataGlobalConstants::CallIndicator::EndZoneSizingCalc);
+
+    // verify no heating or cooling load
+    EXPECT_DOUBLE_EQ(0.0, state->dataSize->CalcZoneSizing(1, 1).HeatLoad);
+    EXPECT_DOUBLE_EQ(0.0, state->dataSize->CalcZoneSizing(1, 1).CoolLoad);
+    // check for correct TstatTemps and ZoneTempsAt*Peaks for output
+    EXPECT_DOUBLE_EQ(22.0, state->dataSize->CalcZoneSizing(1, 1).HeatTstatTemp);
+    EXPECT_DOUBLE_EQ(24.0, state->dataSize->CalcZoneSizing(1, 1).CoolTstatTemp);
+    EXPECT_DOUBLE_EQ(23.0, state->dataSize->FinalZoneSizing(1).ZoneTempAtHeatPeak);
+    EXPECT_DOUBLE_EQ(23.0, state->dataSize->FinalZoneSizing(1).ZoneTempAtCoolPeak);
+
+    state->dataLoopNodes->Node.deallocate();
+    state->dataSize->ZoneEqSizing.deallocate();
+    state->dataHeatBal->Zone.deallocate();
+    state->dataSize->CalcZoneSizing.deallocate();
+    state->dataHeatBalFanSys->NonAirSystemResponse.deallocate();
+    state->dataHeatBalFanSys->SysDepZoneLoads.deallocate();
+    state->dataZoneEquip->ZoneEquipConfig(1).InletNode.deallocate();
+    state->dataZoneEquip->ZoneEquipConfig(1).ExhaustNode.deallocate();
+    state->dataZoneEquip->ZoneEquipConfig.deallocate();
+    state->dataHeatBalFanSys->TempControlType.deallocate();
+    state->dataHeatBalFanSys->TempZoneThermostatSetPoint.deallocate();
+    state->dataHeatBalFanSys->ZoneThermostatSetPointLo.deallocate();
+    state->dataHeatBalFanSys->ZoneThermostatSetPointHi.deallocate();
+    state->dataZoneEnergyDemand->ZoneSysEnergyDemand.deallocate();
+    state->dataZoneEnergyDemand->ZoneSysMoistureDemand.deallocate();
+    state->dataZoneEnergyDemand->DeadBandOrSetback.deallocate();
+    state->dataZoneEnergyDemand->CurDeadBandOrSetback.deallocate();
+    state->dataHeatBalFanSys->ZoneMassBalanceFlag.deallocate();
+    state->dataHeatBal->MassConservation.deallocate();
```

Comment:
I don't know if you need to deellocate() these variables here. I thought the clear_state() function in the various modules already handle it for you.

Reply:
I think you're right, I removed the deallocate() methods.